homework1/task04.py

```diff
@@ -0,0 +1,30 @@
+"""
+Classic task, a kind of walnut for you
+
+Given four lists A, B, C, D of integer values,
+    compute how many tuples (i, j, k, l) there are such that A[i] + B[j] + C[k] + D[l] is zero.
+
+We guarantee, that all A, B, C, D have same length of N where 0 ≤ N ≤ 1000.
+"""
+import itertools
+from typing import List
+
+
+def check_sum_of_four_first_solution(
+    a: List[int], b: List[int], c: List[int], d: List[int]
+) -> int:
+    count = 0
+    for i in a:
+        for j in b:
+            for k in c:
+                for l in d:
+                    if i + j + k + l == 0:
+                        count += 1
+    return count
+
+
+def check_sum_of_four_second_solution(
+    a: List[int], b: List[int], c: List[int], d: List[int]
+) -> int:
+    result = [i for i in itertools.product(a, b, c, d) if sum(i) == 0]
+    return len(result)
```

Comment:
Another option is just `return sum(1 for mumbers in itertools.product(a, b, c, d) if sum(numbers) == 0)